platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemCategory.java

```diff
@@ -1,93 +0,0 @@
-/*
- * Copyright 2023 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.gradle.api.problems.internal;
-
-import com.google.common.base.Objects;
-import com.google.common.collect.ImmutableList;
-import org.gradle.util.Path;
-
-import java.io.Serializable;
-import java.util.Arrays;
-import java.util.List;
-
-/**
- * Default implementation for {@link ProblemCategory}
- */
-public class DefaultProblemCategory implements ProblemCategory, Serializable {
```

Comment:
Dead code... Don't we have some static checkers to fail the build if we have unused classes?

Reply:
no, not for unused classes.

We have for unused private stuff.